addons/mrp/models/mrp_production.py

```diff
@@ -1814,30 +1814,31 @@ def _default_amounts(production):
         assigned_moves = set()
         partially_assigned_moves = set()
         move_lines_to_unlink = set()
+        moves_to_consume = self.env['stock.move']
         for initial_move, backorder_moves in move_to_backorder_moves.items():
             # Create `stock.move.line` for consumed but non-reserved components and for by-products
-            if (initial_move.raw_material_production_id or (initial_move.production_id and initial_move.product_id != production.product_id))\
-                and not initial_move.move_line_ids and set_consumed_qty:
+            if set_consumed_qty and (initial_move.raw_material_production_id or (initial_move.production_id and initial_move.product_id != production.product_id)):
                 ml_vals = initial_move._prepare_move_line_vals()
                 backorder_move_to_ignore = backorder_moves[-1] if has_backorder_to_ignore[initial_move.raw_material_production_id] else self.env['stock.move']
                 for move in list(initial_move + backorder_moves - backorder_move_to_ignore):
```

Comment:
```suggestion
                for move in (initial_move + backorder_moves - backorder_move_to_ignore):
```
Nitpick, but while you're at it, might as well simplify this line, since we don't have to put a recordset in a list to iterate over it.

Reply:
done